@taskgroup decorator implementation  [https://github.com/apache/airflow/issues/11870 ] (#12159)
mistercrunch/Airflow: docs/concepts.rst

.. image:: img/task_group.gif

> @taskgroup decorator

Reviewer: ```suggestion
TaskGroup decorator
```

--------------------

> TaskGroup can be created using @taskgroup decorator, it takes one mandatory argument 'group_id' which is same as constructor of TaskGroup class. It works exactly same as creating TaskGroup using context manager 'with TaskGroup('groupid') as section:'.

Reviewer: ```suggestion
TaskGroup can be created using ``@taskgroup decorator``, it takes one mandatory argument ``group_id`` which is same as constructor of TaskGroup class. It works exactly same as creating TaskGroup using context manager ``with TaskGroup('groupid') as section:``.
```

.. code-block:: python

  @task
  def task_1(value):
      return f'[ Task1 {value} ]'

  @task
  def task_2(value):

>       print( f'[ Task2 {value} ]')

Reviewer: ```suggestion
      print(f'[ Task2 {value} ]')
```

  @taskgroup(group_id='section_1')
  def section_1(value):
      return task_2(task_1(value))

> 

Reviewer: I'm not sure if we need two examples. If we decide to keep both the it would be nice to add some text between them.
